packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -153,6 +154,16 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
     }
   }
 
+  function updateForwardRef(current, workInProgress) {
+    const nextChildren = workInProgress.type.render(
```

Comment:
Let's read `render` first and then call it. Then the fact that it's called with `type` as `this` doesn't become an unintenional part of the public API.

Reply:
Good call.